### DISCUSSION TITLE
「疑问」如何动态更改 page 的样式？

### DISCUSSION THREAD

**Discussion by wenfangdu** [NervJS/taro#15236] [Q & A]
尝试了以下方法，均不生效：

```js
document.querySelector('#app')?.setAttribute('style', 'overflow: hidden')
```

```js
document.querySelector('page')?.setAttribute('style', 'overflow: hidden')
```

```js
document.body.setAttribute('style', 'overflow: hidden')
```

[PageMeta](https://docs.taro.zone/docs/components/page-meta) 试了也没用，Taro 中如何动态更改 page 的样式呢？

**Comment by AdvancedCat** [comment]
可以提供一个 DEMO 哈，应该可以设置的

**Comment by wenfangdu** [comment]
@AdvancedCat 通过以上提到的哪种方法可以设置呢？

**Comment by AdvancedCat** [comment]
应该是都不行的，不同于H5，小程序页面是以 Page 为维度。

可以通过给组件绑定 React Ref 的方式来修改的
```
export default function Index() {

  const ref = useRef(null)

  useLoad(() => {
    console.log('Page loaded.')
    ref.current.setAttribute('style', 'background-color: red;')
  })

  return (
    <View className='index' ref={ref}>
      <Text>Hello world!</Text>
    </View>
  )
}
```

**Comment by TheKonka** [comment]
可以 试下 wx.setPageStyle({complete:console.log, style:{background:'deeppink'}})

**Comment by wenfangdu** [comment]
@TheKonka 请问抖音小程序有类似的 api 嘛？

**Comment by wenfangdu** [comment]
@TheKonka 而且微信小程序文档里没有 `wx.setPageStyle` 呢😅

**Comment by ZakaryCode** [comment]
> 微信小程序文档里没有 `wx.setPageStyle` 呢😅

社区内可以找到[相关的用法](https://developers.weixin.qq.com/community/develop/doc/0000e604898c88d23a3c3686356400)，也可以使用 PageMeta 达到类似效果，但是这些在抖音小程序都不能找到支持依据
